records/generator.go

```diff
@@ -22,16 +22,15 @@ import (
 // Will likely become map[string][]discoveryinfo
 type rrs map[string][]string
 
-// Mesos-DNS state
-// Refactor when discovery id is available
+// RecordGenerator contains DNS records and methods to access and manipulate
+// them. TODO: Refactor when discovery id is available.
 type RecordGenerator struct {
 	As       rrs
 	SRVs     rrs
 	SlaveIPs map[string]string
 }
 
-// The following types help parse state.json
-// Resources holds our SRV ports
+// Resources holds SRV ports.
 type Resources struct {
```

Comment:
I was making all of these type private and realised they need to be public because we have public methods which accept them. See `InsertState` for instance.